static/tests/whisper.js

```diff
@@ -400,71 +196,91 @@ describe('Whisper Tests', function () {
+            var onMessage = function (error, message) {
+                done(error);
+            }
+
             // start watching for messages
-            watchFilter(node2.shh.filter({
-                type: "sym",
+            var params = {
+                symKeyID: keyID2,
                 sig: identity1,
-                key: keyId2,
                 topics: [topic]
-            }), function (err, message) {
-                done(err);
-            });
-
+            }
+            filter = node2.shh.newMessageFilter(params, null, onCreationError);
+            watchFilter(filter, onMessage);
+            
             // send message
             var message = {
-                type: "sym",
+                symKeyID: keyID1,
+                ttl: ttl,
                 sig: identity1,
-                key: keyId1,
                 topic: topic,
                 payload: payload,
-                ttl: 20
+                powTime: powTime,
+                powTarget: powTarget        
             };
-            expect(node1.shh.post(message)).to.equal(null);
+            expect(node1.shh.post(message)).to.equal(true);
         });
-
+        
         it('send asymmetrically encrypted, signed message (node1.id1 -> node2.id2)', function (done) {
-            var payload = 'send asymmetrically encrypted, signed message (node1.id1 -> node2.id2)';
+            this.timeout(0);
+
+            var payload = web3.fromAscii('send asymmetrically encrypted, signed message (node1.id1 -> node2.id2)');
             var topic = makeTopic();
+            var onCreationError = function (error) {
+                done(error);
+            }
+            var onMessage = function (error, message) {
+                done(error);
+            }
+
             // start watching for messages
-            watchFilter(node2.shh.filter({
-                type: "asym",
+            var params = {
+                privateKeyID: identity2,
                 sig: identity1,
-                key: identity2
-            }), function (err, message) {
-                done(err);
-            });
+                topic: topic
+            }
+            filter = node2.shh.newMessageFilter(params, null, onCreationError);
+            watchFilter(filter, onMessage);
 
             // send message
             var message = {
-                type: "asym",
+                pubKey: identity2,
+                ttl: ttl,
                 sig: identity1,
-                key: identity2,
                 topic: topic,
                 payload: payload,
-                ttl: 20
-            };
-            expect(node1.shh.post(message)).to.equal(null);
+                powTime: powTime,
+                powTarget: powTarget        
+            }; 
+            expect(node1.shh.post(message)).to.equal(true);
         });
     });
 
-    context('push notifications', function () {
+    context('chat messages & push notifications', function () {
         this.timeout(5000);
```

Comment:
I disabled the default timeout for the tests which have async tasks since I was running the tests in a very slow internet and missed this one and I also forgot to enable the timeout again for an acceptable time interval. I will apply this change.